### PR TITLE
Update 'check this', 'reveal this' and 'clear this' to work on grouped clues

### PR DESCRIPTION
**Pull request by gtrufitt**
Fixes issue #10529
## Before

before
## After

after

cc/ @NathanielBennett @OliverJAsh @desbo (Nathaniel - you probably know the group stuff better than me, so shout if you can see a cleaner way!)

### CODE REVIEW

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js [hunk 1]
```diff
@@ -375,6 +375,14 @@ class Crossword extends React.Component {
         }
     }
 
+    allHighlightedClues () {
+        return _.filter(this.props.data.entries, (entry) => {
```
Comment (desbo) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: could be simplified to `_.filter(this.props.data.entries, this.clueIsInFocusGroup)`

Comment (gtrufitt) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: Ah nice!

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js [hunk 2]
```diff
@@ -462,7 +470,10 @@ class Crossword extends React.Component {
     }
 
     onCheck () {
-        this.check(this.clueInFocus());
+        // 'Check this' checks single and grouped clues
+        _.forEach(this.allHighlightedClues(), (entry) => {
```
Comment (desbo) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: `_.forEach(this.allHighlightedClues(), this.check)` :)

**Comment by gtrufitt** [comment]
@desbo Updated!

**Comment by desbo** [comment]
nice one! :+1:

**Comment by gtrufitt** [comment]
Updated this to include check this, reveal this and clear this for grouped clues!

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js [hunk 3]
```diff
@@ -492,7 +493,8 @@ class Crossword extends React.Component {
     }
 
     onClearSingle () {
-        const cellsInFocus = helpers.cellsForEntry(this.clueInFocus());
+        // Merge arrays of cells from all highlighted clues
+        const cellsInFocus = [].concat.apply([], _.map(this.allHighlightedClues(), helpers.cellsForEntry, this));
```
Comment (OliverJAsh) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: Could you instead use `_.flatten`?

**Comment by OliverJAsh** [comment]
:+1: Nice!

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js (hunk 3 above)
Comment (gtrufitt) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: :+1: :+1: Yeaaaah!

**Pull request merged**